safety-architecture/tools/callgraph-tool/doc/find_callchains_examples.md

```diff
@@ -0,0 +1,98 @@
+# Instructions for finding call chains in the graph
+
+This page documents instructions and examples on usage of `find_callchains.py` script. The scripts offers a simple way to query all the call chains between one specific function and a set of a functions defined with the regular expression.
+
+```
+./find_callchains.py --help
+usage: find_callchains.py [-h] --calls CALLS --from_function FROM_FUNCTION
+                          --to_function TO_FUNCTION [--out OUT]
+                          [--direction {left,right,both}] [-v]
+
+Find function call chains reachable from the input function. The end of call
+chain is defined with second argument which is a regular expression matching
+one or more function names. The output is either the database of detected call
+chains in csv format or image of the detected subgraph
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --out OUT             name of the output file containing detected chains
+  --direction {left,right,both}
+                        selects search direction.
+  --cutoff CUTOFF       select cutoff lenght for path search
+  -v, --verbose         set the verbosity level (e.g. -vv for debug level)
+
+required named arguments:
+  --calls CALLS         function call database csv file
+  --from_function FROM_FUNCTION
+                        function name from where the search begins (literal
+                        match)
+  --to_function TO_FUNCTION
+                        function name where call chain ends (regex match)
+
+Example: ./find_callchains.py --from_function sock_recvmsg --to_function
+'^x64_sys'--calls calls.csv
+```
+
+
+The script exports the results of a user search either to a `CSV` file (if `out` input parameter ends with .csv) or to an image file otherwise. The source function is always a literal match, while the destination function can be defined as regular expression. The search can be performed in both directions, i.e. both in
+callers of the source function - `left` search and callees of the source function - `right` search.
+
+# Usage examples
+
+
+Suppose that we have an example database for a program with a following call graph:
+
+<img src="all_callchains.jpg">
```

Comment:
label on the 'printf' node in this image seems to have empty strings for the filename and line. Not a big deal, but the node label line seems strange when it contains only the colon ':'. Maybe use 'nan' for empty strings instead, or remove the colon if line and/or filename is empty.

Reply:
Now showing NaN if value is missing